omnibus/files/private-chef-cookbooks/private-chef/libraries/private_chef.rb

```diff
@@ -413,12 +416,117 @@ def gen_frontend
       PrivateChef["bootstrap"]["enable"] = false
     end
 
+    def gen_api_fqdn_default
+      PrivateChef["lb"]["api_fqdn"] ||= PrivateChef['api_fqdn']
+      PrivateChef["lb"]["web_ui_fqdn"] ||= PrivateChef['api_fqdn']
+      PrivateChef["nginx"]["server_name"] ||= PrivateChef['api_fqdn']
+      PrivateChef["nginx"]["url"] ||= "https://#{PrivateChef['api_fqdn']}"
+    end
+
+    def gen_secrets_default(node_name)
+      existing_secrets ||= Hash.new
+      if File.exists?("/etc/opscode/private-chef-secrets.json")
+        existing_secrets = Chef::JSONCompat.from_json(File.read("/etc/opscode/private-chef-secrets.json"))
+      end
+      existing_secrets.each do |k, v|
+        v.each do |pk, p|
+          if not PrivateChef[k]
+            Chef::Log.info("Ignoring unused secret for #{k}.")
```

Comment:
Maybe split out the existing_secrets load to its own function, or do it before the callback and pass it in since a plugin will likely need to use it as well?